Add helm chart schedule-pod-restart (#20)
CoditEU/charts: schedule-pod-restart/README.md

> # Cronjob Pod Restart

Reviewer: ```suggestion
# Schedule Pod Restart
```

Automatically restarting pods with labels based on a cron schedule.

## TL;DR

```console
helm repo add coditeu https://coditeu.github.io/charts
helm repo update

> helm install coditeu/cronjob-pod-restart

Reviewer: ```suggestion
helm install schedule-pod-restart coditeu/schedule-pod-restart
```

```

## Introduction

## Installing the Chart

To install the chart with the release name `schedule-pod-restart`:

```console

> $ helm install schedule-pod-restart coditeu/schedule-pod-restart

Reviewer: Maybe we should give a more representable example where schedule & labels are added?

For example:
```suggestion
$ helm install schedule-pod-restart coditeu/schedule-pod-restart --set "cronjob.schedule=0 1 * * *" --set cronjob.podLabels=app=example
```

Although I haven't tried it so please make sure to test this first 😅

deletes the release.

## Configuration

The following table lists the configurable parameters of the chart and
their default values.

| Parameter                    | Description                                                              | Default                           |
|:-----------------------------|:-------------------------------------------------------------------------|:----------------------------------|
| `nameOverride`               | Name of the cronjob that will be created                                 | Chart Name `schedule-pod-restart` |
| `serviceAccount.name`        | Name of the service account to use                                       | `operations-pod-restart`          |
| `serviceAccount.annotations` | Annotations for the service account                                      | None                              |
| `role.name`                  | Name of the role to create                                               | `manage-pods-role`                |
| `bindings.role.name`         | Name of the role binding                                                 | `manage-pods-role-binding`        |

> | `cronjob.schedule`           | A cronjob expression that defines the schedule when the job should run   | `0 0 * * *`                       |

Reviewer: Maybe add a link to https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax?

```suggestion
| `cronjob.schedule`           | A cronjob expression that defines the schedule when the job should run. Learn more about the format [in the official documentation](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax).   | `0 0 * * *`                       |
```

> | `cronjob.podLabels           | One or more labels on the pods that need to be restarted. Use format `key1=value1,key2=value2`      | None   |

Reviewer: ```suggestion
| `cronjob.podLabels`           | One or more labels on the pods that need to be restarted. Use format `key1=value1,key2=value2`      | None   |
```

> | `bitnamiContainerImageTag`   | bitnami/kubectl container image tag                                      | Version of the Kubernetes cluster where the chart will be deployed          |

Reviewer: Maybe we should go with `image.tag` to align the default approach & not leak that we are using Bitnami?
```suggestion
| `image.tag`   | Tag of container image that contains `kubectl`                                      | Version of the Kubernetes cluster where the chart will be deployed          |
```

Author: Fixed in commit https://github.com/CoditEU/charts/pull/20/commits/807d90c57c6f579bacdf4b908af7e96a4a3a2d35. Note that I couldn't use "tag" (probably a reserved keyword), so changed it to "tagname" => image.tagname

Reviewer: Thanks!
